### DISCUSSION TITLE
Extracting .tgz files

### DISCUSSION THREAD

**Discussion by Tobbe** [saghul/txiki.js#616] [Q&A] [answered]
I have a usecase where I need to download and extract a .tgz file. I looked into a few different npm packages that does "untaring". But they all depend on zlib, which txiki doesn't have (correct me if I'm wrong). 

Is there currently any way to extract .tgz files? What's the best way forward here? Ultimately this is going to be a stand-alone binary. So I need a user-land solution that I can just `tjs compile`.

**Comment by saghul** [comment, marked as answer]
The compression streams use a JS implementation of gz but it's not directly exposed, I guess you'd need to pipe the stream and untar on the other end.

I'd like txiki to ship with bindings to miniz for fast (de)compression and expose it as an API, but no ETA yet 😅

**Reply by Tobbe** [reply]
Sounds good. I guess I'll wait for the miniz bindings then 🙂
